addons/payment/views/portal_templates.xml

```diff
@@ -142,18 +142,109 @@
                 <div class="container">
                     <!-- Portal breadcrumb -->
                     <t t-call="payment.portal_breadcrumb"/>
-                    <!-- Status page -->
+                    <!-- Payment status page -->
                     <div name="o_payment_status" class="row justify-content-center my-3">
-                        <div name="o_payment_status_content"
-                             class="col-12 col-lg-8">
-                            <!-- The content is generated in JavaScript -->
+                        <div name="o_payment_status_content" class="col-12 col-lg-8">
+                            <t t-call="payment.transaction_details"/>
                         </div>
                     </div>
                 </div>
             </div>
         </t>
     </template>
 
+    <template id="payment.payment_not_found" name="Payment Not Found">
+        <t t-call="portal.frontend_layout">
+            <t t-set="page_title" t-value="'Payment Not Found'"/>
+            <t t-set="additional_title"><t t-esc="page_title"/></t>
+            <div class="wrap">
+                <div class="container">
+                    <!-- Portal breadcrumb -->
+                    <t t-call="payment.portal_breadcrumb"/>
+                    <!-- Payment not found page -->
+                    <div name="o_payment_not_found" class="row justify-content-center my-3">
+                        <div name="o_payment_not_found_content" class="col-12 col-lg-8 text-center">
+                            <p>We're unable to find your payment, but don't worry.</p>
+                            <p>
+                                You should receive an email confirming your payment in a few
+                                minutes.
+                            </p>
+                            <p>If the payment hasn't been confirmed you can contact us.</p>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </t>
+    </template>
+
+    <template id="payment.transaction_details">
```

Comment:
This template is similar to `payment.transaction_status` so I wonder if we should create a task to extract the duplicated bits. This would likely require a review of the state messages and icons as they differ slightly.

Reply:
Yes, and I even believe that you could do it in this PR without too much hassle if you
1. move the payment summary from `payment.transaction_details` to its parent template `payment.payment_status`;
2. make `payment.transaction_status` more generic so that it can:
  a. determine `alert_style`, `alert_header`, `alert_message` based on the received `tx`;
  b. remove `display_message` from `payment.transaction._get_post_processing_values`;
  c. display an alert block with the determined appropriate style and content;
  d. accept configuration keys to adapt the icon and display the "Skip" button.
3. make `payment.payment_status` call `payment.transaction_status`;
4. delete `payment.transaction_details`;
5. rename `payment.transaction_status` into `payment.state_header` so we finally have a good name for it.

Reply:
![image](https://github.com/odoo/odoo/assets/15896813/01d1c702-1dfd-4401-a5c8-4ad5ccc4daac)

https://github.com/odoo/odoo/blob/master/addons/website_sale/views/templates.xml#L2467-L2487

Reply:
FYI, this change also impacts enterprise, see https://github.com/odoo/enterprise/pull/58822

Reply:
I just finished retesting, everything looks fine :)